src/generate-hyperlinks.js

```diff
@@ -0,0 +1,49 @@
+//const url = 'https://docs.google.com/spreadsheets/d/1NAhJflgwX2BGhJG39Wg9vOhwB_85kaJPt5pWzHY5bmA/edit#gid=1193677007';
+
+// f.generating links to a given cell
+function getLink(range,sheetName) {
+  const spreadsheet = SpreadsheetApp.getActiveSpreadsheet();
+  const sheet = spreadsheet.getSheetByName(sheetName);
+  const link = "https://docs.google.com/spreadsheets/d/"+spreadsheet.getId()+"/edit#gid="+sheet.getSheetId()+"&range="+range.getA1Notation();
+  return link;
+}
+
+// f. finding in the 21st column a cell with a given word
+function find (word,inSheet) {
```

Comment:
Ta funkcja jest do przepisania. Analizowałem w głowie czemu to działa tak wolno, no i problemem jest że dla każdego słowa lecisz po całym spreadsheecie szukając dla niego odpowiednika. Zrób w taki sposób:

1. Przygotuj do zmiennej mapę, która będzie mapowała słowo na jej indeks(range) w atergo (czas O(n))
2. następnie wykorzystaj tą mapę (mapę, słownik, jak zwał tak zwał) żeby wyciągać range każdego słowa. czas O(1)
3. pkt. 2. musimy powtórzyć n razy, więc będziemy mieli czas O(2n).

obecne rozwiązanie ma czas O(n^2), stąd jest wolne. :)

Jestem przekonany, że dla 10000 rzędów przy takim rozwiązaniu uda nam się odświeżyć całą listę szybicej niż w minutę :D

Reply:
Totally agree, came to my mind while writing it, but just wanted to make it work first. I will try. (to be done)